src/libpmemfile-posix/fcntl.c

```diff
@@ -102,8 +107,61 @@ pmemfile_fcntl(PMEMfilepool *pfp, PMEMfile *file, int cmd, ...)
 			else if ((file->flags & PFILE_WRITE) == PFILE_WRITE)
 				ret |= PMEMFILE_O_WRONLY;
 
+fl_end:
+			os_mutex_unlock(&file->mutex);
 			return ret;
 		}
+		case PMEMFILE_F_SETFL:
+		{
+			va_list ap;
+			va_start(ap, cmd);
+			int fl_flags = va_arg(ap, int);
+			va_end(ap);
+
+			fl_flags &= ~(PMEMFILE_O_ACCMODE | PMEMFILE_O_CREAT |
+					PMEMFILE_O_EXCL | PMEMFILE_O_NOCTTY |
+					PMEMFILE_O_TRUNC | PMEMFILE_O_SYNC |
+					PMEMFILE_O_DSYNC);
+
+			if (fl_flags & PMEMFILE_O_ASYNC) {
+				ERR("setting O_ASYNC is not supported");
+				errno = EINVAL;
+				return -1;
+			}
+
+#define SUPPORTED_FLAGS (PMEMFILE_O_APPEND | PMEMFILE_O_DIRECT | \
+		PMEMFILE_O_NOATIME | PMEMFILE_O_NONBLOCK)
+
+			if (fl_flags & ~SUPPORTED_FLAGS) {
+				ERR("unknown flag %d",
+						fl_flags & ~SUPPORTED_FLAGS);
+				errno = EINVAL;
+				return -1;
+
+			}
+#undef SUPPORTED_FLAGS
+
+			os_mutex_lock(&file->mutex);
+
+			if (fl_flags & PMEMFILE_O_APPEND)
+				file->flags |= PFILE_APPEND;
```

Comment:
This changes are shared among dup'ed fds.
So, to be clear, these are the *file status flags*, shared by dup'ed fd's, and the *file descriptor flags* are not shared, and those should be stored in the libpmemfile layer. Right?
And the only *file descriptor flag* we know about is the close-on-exec flag, if I interpret things right.

So, `F_SETFD` should be handled in libpmemfile, and not passed on to libpmemfile-posix, I guess.

Reply:
You are right, but it's not relevant to this patch :). This patch is about F_SETFL, not F_SETFD.

Reply:
When I looked at it, I just wasn't sure if these changes should be OK with dup, or not, but if you agree too, then OK.